linux_new/Jenkinsfile

```diff
@@ -0,0 +1,959 @@
+        error("Checksum mismatch for ${file}. Expected: ${expectedChecksum}, but found: ${calculatedChecksum}.")
+    }
+    echo "Checksum validation successful for ${file}."
+    return calculatedChecksum
+}
+
+// Helper function to extract upstream version from tar file
+def extractUpstreamVersionFromTar(file) {
+    def version = sh(script: "tar xvfz \"${file}\" | head -n 1 | cut -d/ -f1", returnStdout: true).trim()
+    sh "rm -rf \"${version}\""
+    echo "Extracted upstream version: ${version}"
+    return version
+}
+
+// Helper function to cleanup files
+def cleanupFiles(files) {
+    files.each { file ->
+        sh "rm -rf ${file}"
+    }
+}
+
+// Helper function to Set Node Labels
+def getPackageBuildLabel(String arch, String distro) {
+    switch (distro) {
+        case 'APK':
+            return (arch == 'x64') ? 'build&&linux&&x64&&dockerBuild&&dynamicAzure' :
+                   (arch == 'aarch64') ? 'docker&&linux&&aarch64&&alpineInstaller' :
+                   'build&&linux&&x64&&dockerBuild&&dynamicAzure'
+
+        case 'DEB':
+            return (arch == 'x64') ? 'build&&linux&&x64&&dockerBuild&&dynamicAzure' :
+                   (arch == 'arm') ? 'docker&&linux&&aarch64' :
+                   (arch == 'aarch64') ? 'docker&&linux&&aarch64' :
+                   (arch == 'ppc64le') ? 'build&&docker&&ppc64le' :
+                   (arch == 's390x') ? 'docker&&s390x&&dockerBuild' :
+                   (arch == 'riscv64') ? 'dockerBuild&&linux&&riscv64&&dockerInstaller' :
+                   
+                   'build&&linux&&x64&&dockerBuild&&dynamicAzure'
+
+        case 'RHEL':
+            return 'build&&linux&&x64&&dockerBuild&&dynamicAzure'
+
+        default:
+            error("Unsupported distro: ${distro}")
+    }
+}
+
+// Helper function to check and upload files
+def CheckAndUpload(String Target, String Distro, String BuildArch, String RelVersion, String DistroList, String Value, String PackageDir, String Key, String FileName) {
+  echo "Entering Check & Upload"
+  // Define A Context Appropriate Release Number
+  def RelNumber = params.VERSION.replaceAll(/[^\d]/, '')
+
+  echo "Debug 2001"
+  echo "${Target} ${Distro} ${BuildArch} ${RelVersion} ${DistroList} ${Value} ${PackageDir} ${Key} ${FileName}"
+  echo "Target = ${Target}/${FileName}"
+  echo "Distro = ${Distro}"
+  echo "Arch = ${BuildArch}"
+  echo "Version = ${RelVersion}"
+  echo "RELNUM = ${RelNumber}"
+  echo "DistroList = ${DistroList}"
+  echo "Value = ${Value}"
+  echo "PackageDir = ${PackageDir}"
+  echo "Key = ${Key}"
+  echo "Filename = ${FileName}"
+
+  env.TARGET = Target
+  env.DISTRO = Distro
+  env.BUILDARCH = BuildArch
+  env.VERSION = RelVersion
+  env.RELNUM = RelNumber
+  env.DISTROLIST = DistroList
+  env.VALUE = Value
+  env.PACKAGEDIR = PackageDir
+  env.KEY = Key
+  env.FILENAME = FileName
+  // env.DESTTARGET = RHELTarget = "${RHELkey}/Packages"
+  def artBaseURL = "https://adoptium.jfrog.io/artifactory/"
```

Comment:
Should this not be based on the `SERVER_ID` job parameter?

Reply:
I noticed this was missing when I copied the original, so I've removed the parameter..  there'd be a few more changes should we ever move away from artifactory, so  I think living the constant in the jenkinsfile will do for the moment. The actual upload is dependent on the jenkins / artifactory plugin and its configuration.

Reply:
Yep I'm good with that - just didn't want confusion caused by it being in two places :-)